internal/ports/storage/order.go

```diff
@@ -0,0 +1,259 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/jackc/pgerrcode"
+	"github.com/lib/pq"
+	"github.com/shopspring/decimal"
+	"github.com/shulganew/gophermart/internal/entities"
+)
+
+func (r *Repo) AddOrder(ctx context.Context, userID uuid.UUID, order string, isPreOrder bool, withdrawn decimal.Decimal) error {
```

Comment:
```suggestion
func (r *Repo) AddOrder(ctx context.Context, data entities.AddOrder) error {
```
лучше не оставлять большое количество аргументов - можно упаковать их в одну DTO